actioncable/lib/action_cable/channel/base.rb

```diff
@@ -153,6 +155,14 @@ def initialize(connection, identifier, params = {})
         subscribe_to_channel
       end
 
+      # This method is called after subscription has been added to the channel.
+      # Send confirmation here to avoid race conditions when client tries to perform actions
+      # right after receiving confirmation.
+      def registered!
```

Comment:
We can go with the existing `subscribe_to_channel`:
- remove `subscribe_to_channel` call from `initialize';
- make `subscribe_to_channel` public;
- call `subscription.subscribe_to_channel` in `Connection::Subscriptions#add`.

It make sense, 'cause we already have `Channel::Base#unsubscribe_from_channel` public method.